QE.md

```diff
@@ -0,0 +1,103 @@
+|  Quantum ESPRESSO | 6.7.0  |  https://github.com/QEF/q-e/archive/refs/tags/qe-6.7.0.tar.gz |
+### OS Requirements
+| Item  | Version  | How to Obtain  |
+| ------------ | ------------ | ------------ |
+|  CentOS | 8.0  |  https://www.centos.org/download/ |
+| Kernel  | 4.18.0-80  |  Included in the OS image. |
+## Configuring the Compilation Environment
+
+### Installing GNU 9.3
+
+
+    yum install -y centos-release-scl
+    yum install -y devtoolset-9-gcc
+    yum install -y devtoolset-9-gcc-c++
+    yum install -y devtoolset-9-binutils
+    scl enable devtoolset-9 bash
+    echo "source /opt/rh/devtoolset-9/enable" >> /etc/profile
+### Installing Open MPI
+1. Run the following command to install the system dependency package:
+```
+    yum install libxml2* systemd-devel.aarch64 numa* -y
+```
+2. Run the following commands to install Open MPI:
+```
+    wget https://download.open-mpi.org/release/open-mpi/v4.0/openmpi-4.0.3.tar.gz
+    tar -zxvf openmpi-4.0.3.tar.gz
+    cd openmpi-4.0.3
+    ./configure --prefix=/path/to/OPENMPI --enable-pretty-print-stacktrace --enable-orterun-prefix-by-default  --with-cma --enable-mpi1-compatibility
+    make -j 16
+    make install
+```
+3. Configure environment variables:
+
+```
+    export PATH=/path/to/OPENMPI/bin:$PATH
+    export LD_LIBRARY_PATH=/path/to/OPENMPI/lib:$LD_LIBRARY_PATH
+```
+### Installing openblas
+1. Run the following commands to install OpenBLAS:
+```
+    wget -O OpenBLAS-0.3.14.tar.gz https://github.com/xianyi/OpenBLAS/archive/refs/tags/v0.3.14.tar.gz 
+    tar -zxvf OpenBLAS-0.3.14.tar.gz
+    cd OpenBLAS-0.3.14
+	export CC=`which gcc`
+	export CXX=`which g++`
+	export FC=`which gfortran`
+    make
+    make PREFIX=/path/to/OPENBLAS install
+```
+2. Configure environment variables:
+```
+    export LD_LIBRARY_PATH=/path/to/OPENBLAS/lib:$LD_LIBRARY_PATH
+```
+### Installing Scalapack
+1. Run the following commands to install Scalapack:
+
+```
+    wget http://www.netlib.org/scalapack/scalapack-2.1.0.tgz
+    tar -xvf scalapack-2.1.0.tgz
+    cd scalapack-2.1.0
+    cp SLmake.inc.example SLmake.inc
+	vi SLmake.inc ,and set:
+	BLASLIB       = -L/path/to/OPENBLAS/lib -lopenblas
+	LAPACKLIB     = -L/path/to/OPENBLAS/lib -lopenblas
+    make
+    cp libscalapack.a /path/to/SCALAPACK
+```
+2. Configure environment variables:
+```
+	export LD_LIBRARY_PATH=/path/to/SCALAPACK:$LD_LIBRARY_PATH
+```
+### Installing Quantum ESPRESSO
+1. Run the following commands to install QE:
+```
+    wget https://github.com/QEF/q-e/archive/refs/tags/qe-6.7.0.tar.gz
+    tar -zxvf qe-6.7.0.tar.gz
+    cd q-e-qe-6.7.0
+    export BLAS_LIBS="-L/path/to/OPENBLAS/lib -lopenblas"
+	export LAPACK_LIBS="-L/path/to/OPENBLAS/lib -lopenblas"
+	export SCALAPACK_LIBS="-L/path/to/SCALAPACK -lscalapack"
+	./configure F90=gfortran F77=gfortran MPIF90=mpifort MPIF77=mpifort CC=mpicc \
+	FCFLAGS="-O3" CFLAGS="-O3" \
+	--with-scalapack=yes \
+	--prefix=/path/to/QE
+	make -j20 pwall
```

Comment:
use `-j $(nproc)` instead ?!

Reply:
`make -j20 pwall` fails for me with:

```
mpifort -O3 -g -cpp -D__FFTW -D__MPI -D__SCALAPACK   -I/WRF/q-e-qe-6.7.0//include -I/WRF/q-e-qe-6.7.0//FoX/finclude   -c xmltools.f90
mpifort -O3 -g -cpp -D__FFTW -D__MPI -D__SCALAPACK   -I/WRF/q-e-qe-6.7.0//include -I/WRF/q-e-qe-6.7.0//FoX/finclude   -c upf_invmat.f90
xmltools.f90:852.18:

       cp = c_loc(zmat)
                  1
Error: Assumed-shape array 'zmat' at (1) cannot be an argument to the procedure 'c_loc' because it is not C interoperable
xmltools.f90:826.18:

       cp = c_loc(zvec)
                  1
Error: Assumed-shape array 'zvec' at (1) cannot be an argument to the procedure 'c_loc' because it is not C interoperable
xmltools.f90:511.15:

    cp = c_loc(zmat)
               1
Error: Assumed-shape array 'zmat' at (1) cannot be an argument to the procedure 'c_loc' because it is not C interoperable
xmltools.f90:487.15:

    cp = c_loc(zvec)
               1
Error: Assumed-shape array 'zvec' at (1) cannot be an argument to the procedure 'c_loc' because it is not C interoperable
make[1]: *** [xmltools.o] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: Leaving directory `/WRF/q-e-qe-6.7.0/upflib'
make: *** [libupf] Error 1
make: *** Waiting for unfinished jobs....
...
```

Reply:
I was able to update gfortran to 7.3.1 by following the instructions from https://forums.centos.org/viewtopic.php?t=68954
```
yum install devtoolset-7-toolchain
scl enable devtoolset-7 bash
$ gfortran --version | head -2
GNU Fortran (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
```
but still the error persist.

Reply:
I've tried with devtoolset-10 (gfortran 10.2.1 20210130) and this overcame the problem but now it fails later with:
```
 ...
	bands.o libpp.a ../../PW/src/libpw.a ../../Modules/libqemod.a ../../upflib/libupf.a ../../KS_Solvers/libks_solvers.a ../../FFTXlib/libqefft.a ../../LAXlib/libqela.a ../../UtilXlib/libutil.a ../../dft-d3/libdftd3qe.a /WRF/q-e-qe-6.7.0//clib/clib.a /WRF/q-e-qe-6.7.0//LIBBEEF/libbeef.a   -L/path/to/SCALAPACK -lscalapack -L/path/to/OPENBLAS/lib -lopenblas -L/WRF/q-e-qe-6.7.0//FoX/lib  -lFoX_dom -lFoX_sax -lFoX_wxml -lFoX_common -lFoX_utils -lFoX_fsys   -L/path/to/OPENBLAS/lib -lopenblas    
/opt/rh/devtoolset-10/root/usr/libexec/gcc/aarch64-redhat-linux/10/ld: ../../LAXlib/libqela.a(cdiaghg.o): in function `laxlib_pcdiaghg_':
/WRF/q-e-qe-6.7.0/LAXlib/cdiaghg.f90:575: undefined reference to `pzpotrf_'
/opt/rh/devtoolset-10/root/usr/libexec/gcc/aarch64-redhat-linux/10/ld: /WRF/q-e-qe-6.7.0/LAXlib/cdiaghg.f90:598: undefined reference to `pztrtri_'
/opt/rh/devtoolset-10/root/usr/libexec/gcc/aarch64-redhat-linux/10/ld: ../../LAXlib/libqela.a(rdiaghg.o): in function `laxlib_prdiaghg_':
/WRF/q-e-qe-6.7.0/LAXlib/rdiaghg.f90:513: undefined reference to `pdpotrf_'
/opt/rh/devtoolset-10/root/usr/libexec/gcc/aarch64-redhat-linux/10/ld: /WRF/q-e-qe-6.7.0/LAXlib/rdiaghg.f90:533: undefined reference to `pdtrtri_'
/opt/rh/devtoolset-10/root/usr/libexec/gcc/aarch64-redhat-linux/10/ld: ../../LAXlib/libqela.a(dspev_drv.o): in function `__dspev_module_MOD_pdsyevd_drv':
/WRF/q-e-qe-6.7.0/LAXlib/dspev_drv.f90:723: undefined reference to `pdsyevd_'
/opt/rh/devtoolset-10/root/usr/libexec/gcc/aarch64-redhat-linux/10/ld: /WRF/q-e-qe-6.7.0/LAXlib/dspev_drv.f90:733: undefined reference to `pdsyevd_'
/opt/rh/devtoolset-10/root/usr/libexec/gcc/aarch64-redhat-linux/10/ld: /WRF/q-e-qe-6.7.0/LAXlib/dspev_drv.f90:723: undefined reference to `pdsyevd_'
/opt/rh/devtoolset-10/root/usr/libexec/gcc/aarch64-redhat-linux/10/ld: /WRF/q-e-qe-6.7.0/LAXlib/dspev_drv.f90:733: undefined reference to `pdsyevd_'
/opt/rh/devtoolset-10/root/usr/libexec/gcc/aarch64-redhat-linux/10/ld: /WRF/q-e-qe-6.7.0/LAXlib/dspev_drv.f90:733: undefined reference to `pdsyevd_'
/opt/rh/devtoolset-10/root/usr/libexec/gcc/aarch64-redhat-linux/10/ld: ../../LAXlib/libqela.a(dspev_drv.o):/WRF/q-e-qe-6.7.0/LAXlib/dspev_drv.f90:723: more undefined references to `pdsyevd_' follow
( cd ../../bin ; ln -fs ../PP/src/average.x . )
/opt/rh/devtoolset-10/root/usr/libexec/gcc/aarch64-redhat-linux/10/ld: ../../LAXlib/libqela.a(zhpev_drv.o): in function `__zhpev_module_MOD_pzheevd_drv':
/WRF/q-e-qe-6.7.0/LAXlib/zhpev_drv.f90:1547: undefined reference to `pzheevd_'
/opt/rh/devtoolset-10/root/usr/libexec/gcc/aarch64-redhat-linux/10/ld: /WRF/q-e-qe-6.7.0/LAXlib/zhpev_drv.f90:1560: undefined reference to `pzheevd_'
/opt/rh/devtoolset-10/root/usr/libexec/gcc/aarch64-redhat-linux/10/ld: /WRF/q-e-qe-6.7.0/LAXlib/zhpev_drv.f90:1547: undefined reference to `pzheevd_'
/opt/rh/devtoolset-10/root/usr/libexec/gcc/aarch64-redhat-linux/10/ld: mpifort -g -o dos.x \
	dos.o libpp.a ../../PW/src/libpw.a ../../Modules/libqemod.a ../../upflib/libupf.a ../../KS_Solvers/libks_solvers.a ../../FFTXlib/libqefft.a ../../LAXlib/libqela.a ../../UtilXlib/libutil.a ../../dft-d3/libdftd3qe.a /WRF/q-e-qe-6.7.0//clib/clib.a /WRF/q-e-qe-6.7.0//LIBBEEF/libbeef.a   -L/path/to/SCALAPACK -lscalapack -L/path/to/OPENBLAS/lib -lopenblas -L/WRF/q-e-qe-6.7.0//FoX/lib  -lFoX_dom -lFoX_sax -lFoX_wxml -lFoX_common -lFoX_utils -lFoX_fsys   -L/path/to/OPENBLAS/lib -lopenblas    
/WRF/q-e-qe-6.7.0/LAXlib/zhpev_drv.f90:1560: undefined reference to `pzheevd_'
/opt/rh/devtoolset-10/root/usr/libexec/gcc/aarch64-redhat-linux/10/ld: /WRF/q-e-qe-6.7.0/LAXlib/zhpev_drv.f90:1560: undefined reference to `pzheevd_'
/opt/rh/devtoolset-10/root/usr/libexec/gcc/aarch64-redhat-linux/10/ld: ../../LAXlib/libqela.a(zhpev_drv.o):/WRF/q-e-qe-6.7.0/LAXlib/zhpev_drv.f90:1547: more undefined references to `pzheevd_' follow
collect2: error: ld returned 1 exit status
make[2]: *** [Makefile:80: open_grid.x] Error 1
make[2]: *** Waiting for unfinished jobs....
```